SofaKernel/framework/sofa/core/objectmodel/Link.h

```diff
@@ -555,7 +555,8 @@ class TLink : public BaseLink
             }
 
             // Remove the objects from the container that are not in the new list
-            for (size_t i = 0; i != container.size(); i++)
+            unsigned int csize = (unsigned int)container.size();
+            for (unsigned int i = 0; i != csize; i++)
```

Comment:
What about `for (size_t i = 0, n = container.size() ; i != n ; i++)` ?

Reply:
Yes I wanted to discuss that. Because it is a problem we have everywhere in Sofa.
if i is a site_t, then in the next line:
DestPtr dest(container[i]);
container should be of <size_t> which has a big impact. Nearly a replace all uint into size_t in objectModel::Link BaseLink etc... 

You find the same problem in topology mapping where the containers Map<unsigned int, unsigned int> are holding the id of buffers with a max of size_t.

Reply:
🦈 🦈 🦈